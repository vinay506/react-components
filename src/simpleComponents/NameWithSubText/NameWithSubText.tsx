/**
 * 
 * Descirption:
 *   Its a Wrapper of Name and its sub text
 *  */ 

import React from 'react'
import styled from 'styled-components';

const NameWithSubText = (props: any) => {
    return (
        <Name>
            <div className="nameText">{props.nameInfo.name}</div>
            <div className="subText">{props.nameInfo.subText}</div>
        </Name>
    )
}


const Name = styled.div`
    color:#aecbfa;
    font-size:12px;
`


export default NameWithSubText;
