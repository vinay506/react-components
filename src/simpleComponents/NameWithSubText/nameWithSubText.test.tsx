import React from 'react';
import { shallow } from 'enzyme';
import NameWithSubText from './NameWithSubText';
import renderer from 'react-test-renderer'
import 'jest-styled-components'


describe('<Name with subText>', function () {
    let wrapper:any;
    let props:any;
    beforeEach(() => {
        props = {
            nameInfo: {
                name: 'Asha Sharma',
                subText: '60 AIR'
            }
        };
        wrapper = shallow(<NameWithSubText nameInfo={props.nameInfo} ></NameWithSubText>);
    });

    // Snapshot test
    it('Name with subtext snapshot', () => {
        const tree = renderer.create(<NameWithSubText nameInfo={props.nameInfo}></NameWithSubText>).toJSON()
        expect(tree).toMatchSnapshot(`
        .c0 {
            color:#aecbfa;
            font-size:12px;
        }
        <div className='sc-fzozJi ictDee'
        >
            <div className="nameText">Asha Sharma</div>
            <div className="subText">60 AIR</div>
        </div>`)
    })

    it('renders Text of Name', () => {
        const nameText = wrapper.find('.nameText').text();
        expect(nameText).toEqual(props.nameInfo.name);
    });

    it('renders Sub Text of Name', () => {
        const nameText = wrapper.find('.subText').text();
        expect(nameText).toEqual(props.nameInfo.subText);
    });
})