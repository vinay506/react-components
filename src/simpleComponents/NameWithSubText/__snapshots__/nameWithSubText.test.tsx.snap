// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Name with subText> Name with subtext snapshot: 
        .c0 {
            color:#aecbfa;
            font-size:12px;
        }
        <div className='sc-fzozJi ictDee'
        >
            <div className="nameText">Asha Sharma</div>
            <div className="subText">60 AIR</div>
        </div> 1`] = `
.c0 {
  color: #aecbfa;
  font-size: 12px;
}

<div
  className="c0"
>
  <div
    className="nameText"
  >
    Asha Sharma
  </div>
  <div
    className="subText"
  >
    60 AIR
  </div>
</div>
`;
